Editor/Scripts/Analytics/MeshsyncAnalytics.cs

```diff
@@ -91,11 +157,84 @@ private struct SyncEventData {
+            {
+                LogTime = DateTimeOffset.Now,
+                Data = eventData
+            });
+
+            if (!CoalesceEvents) {
+                var evts = this.syncEventStack.Select(evt => evt.Data).ToArray();
+
+                foreach (var evt in evts) {
+                    logIfWarning(
+                        EditorAnalytics.SendEventWithLimit(
+                            MESHSYNC_SYNC_INSTALLEDEVENTNAME,
+                            evt,
+                            MESHSYNC_SYNC_VERSION));
+                }
+                this.syncEventStack.Clear();
+            }
+            else if ((DateTime.Now - this.lastSync) > MinTimeBetweenSync) {
+
+                // Group entity sync events by type
+                var syncEntityTypeLookup = this.syncEventStack
+                    .Where(x => x.Data.entitySyncType != "none")
+                    .ToLookup(x => x.Data.entitySyncType);
+
+                // Group asset sync events by type
+                var syncAssetTypeLookup = this.syncEventStack
+                    .Where(x => x.Data.assetSyncType != "none")
+                    .ToLookup(x => x.Data.assetSyncType);
+
+                // Coalesce events by time
+                var syncEntityTypeFinalBatch = CoalesceEventsByTime(syncEntityTypeLookup);
+                var syncAssetTypeFinalBatch = CoalesceEventsByTime(syncAssetTypeLookup);
+
+                var combined = new List<TimedEvent<SyncEventData>>();
+
+                combined.AddRange(syncEntityTypeFinalBatch);
+                combined.AddRange(syncAssetTypeFinalBatch);
+
+                // Re-order combined events by time
+                var sorted = combined
+                    .OrderBy(evt => evt.LogTime)
+                    .Select(evt => evt.Data);
+
+                foreach (var evt in sorted) {
+                    logIfWarning(
+                        EditorAnalytics.SendEventWithLimit(
+                            MESHSYNC_SYNC_INSTALLEDEVENTNAME,
+                            evt,
+                            MESHSYNC_SYNC_VERSION));
+
+                }
+
+                this.lastSync = DateTime.Now;
+
+                this.syncEventStack.Clear();
+
+            }
+        }
+
+        /// <summary>
+        /// Groups up events by time interval and picks one per second
+        /// </summary>
+        /// <typeparam name="T">Event data type</typeparam>
+        /// <param name="events">Events grouped by type</param>
+        /// <returns>Flattened list of events limited to one per second per event type</returns>
+        private static IEnumerable<TimedEvent<T>> CoalesceEventsByTime<T>(ILookup<string, TimedEvent<T>> events)
+            where T: struct {
+            var list = new List<TimedEvent<T>>();
+
+            foreach (var group in events) {
+                var perSecond = group.GroupBy(evt => evt.LogTime.ToUnixTimeSeconds())
+                    .Select(x => x.First());
+
+                list.AddRange(perSecond);
```

Comment:
It seems to me that this code will still put the same amount of events in the list.
It's just that, instead of all events, this code will put the first event in that second to the list as many as the number of its occurrence.
Is that expected ?

I would recommend to create a simple unit test for this function.

Reply:
This looks correct to me, but a unit test might be good to have in any case. If [this](https://github.com/unity3d-jp/MeshSync/pull/807/files#r1004424231) is implemented, I think this function can be removed.